File: day02/js/solution.js
import fs from "fs";

const input = fs.readFileSync("../input.txt", "utf-8").trim();

function isInvalidID(n) {
  const s = n.toString();
  const len = s.length;

  if (len % 2 !== 0) return false;

  const half = len / 2;
  return s.slice(0, half) === s.slice(half);
}

function solve(input) {
  let sum = 0;
  const ranges = input.split(",");

  for (const r of ranges) {
    const [start, end] = r.split("-").map(Number);

    for (let n = start; n <= end; n++) {
      if (isInvalidID(n)) {
        sum += n;
      }
    }
  }

  return sum;
}

console.log("Result:", solve(input));

File: day02/py/solution.py
from pathlib import Path

input_data = Path("../input.txt").read_text().strip()

def is_invalid_id(n):
    s = str(n)
    if len(s) % 2 != 0:
        return False

    half = len(s) // 2
    return s[:half] == s[half:]

def solve(data):
    total = 0
    ranges = data.split(",")

    for r in ranges:
        start, end = map(int, r.split("-"))

        for n in range(start, end + 1):
            if is_invalid_id(n):
                total += n

    return total

print("Result:", solve(input_data))
